src/main/java/org/folio/rest/impl/InvoiceHelper.java

```diff
@@ -149,15 +177,340 @@
+   * and exist at least one {@link InvoiceLine} associated with this invoice
+   *
+   * @param invoice {@link Invoice}to be approved
+   * @return CompletableFuture that indicates when transition is completed
+   */
+  private CompletableFuture<Void> approveInvoice(Invoice invoice) {
+    CompletableFuture<Void> future = new VertxCompletableFuture<>(ctx);
+    getInvoiceLinesWithTotals(invoice)
+      .thenApply(invoiceLines -> {
+        verifyInvoiceLineNotEmpty(invoiceLines);
+        return invoiceLines;
+      })
+      .thenCombine(prepareVoucher(invoice), (invoiceLines, voucher) -> {
+        invoice.setVoucherNumber(voucher.getVoucherNumber());
+        return handleVoucherWithLines(invoiceLines, voucher)
+          .thenAccept(future::complete)
+          .exceptionally(t -> {
+            future.completeExceptionally(t);
+            return null;
+          });
+      })
+    .exceptionally(t -> {
+      future.completeExceptionally(t);
+      return null;
+    });
+    return future;
+  }
+
+  private void verifyInvoiceLineNotEmpty(List<InvoiceLine> invoiceLines) {
+    if (invoiceLines.isEmpty()) {
+      throw new HttpException(500, NO_INVOICE_LINES_ERROR_MSG);
+    }
+  }
+
+  /**
+   * Prepares a new voucher or updates existing one for further processing
+   *
+   * @param invoice {@link Invoice} to be approved on the basis of which the voucher is prepared
+   * @return completable future with {@link Voucher} on success
+   */
+  private CompletableFuture<Voucher> prepareVoucher(Invoice invoice) {
+    return findExistingVoucher(invoice.getId())
+      .thenCompose(vouchers -> {
+        if (isNotEmpty(vouchers)) {
+          Voucher voucher = vouchers.get(0);
+          return updateExistingVoucherState(invoice, voucher);
         }
+        return buildNewVoucher(invoice);
       });
   }
 
+  private CompletableFuture<List<Voucher>> findExistingVoucher(String invoiceId) {
+    return voucherHelper.getVouchers(1, 0, String.format(QUERY_BY_INVOICE_ID, invoiceId))
+      .thenApply(VoucherCollection::getVouchers);
+  }
+
+  /**
+   * Updates state of existing {@link Voucher} linked with processed {@link Invoice}
+   *
+   * @param invoice invoice {@link Invoice} to be approved
+   * @param voucher {@link Voucher} from voucher-storage related to processed invoice
+   * @return completable future with {@link Voucher}
+   */
+  private CompletableFuture<Voucher> updateExistingVoucherState(Invoice invoice, Voucher voucher) {
+    voucher.setInvoiceCurrency(invoice.getCurrency());
+    setDefaultRequiredFields(voucher);
+
+    return VertxCompletableFuture.completedFuture(voucher);
+  }
+
+  //TODO Start using real information to create a voucher when it becomes known where to get these values from.
+  private void setDefaultRequiredFields(Voucher voucher) {
+    voucher.setAccountingCode(DEFAULT_ACCOUNTING_CODE);
+    voucher.setExportToAccounting(false);
```

Comment:
I believe you can use latest version of `acq-models` to get `exportToAccounting` value from `invoice` as per [Acquisitions Interface Fields - Voucher](https://docs.google.com/spreadsheets/d/1F88PoSnEKD7oFnza1ZuOgzAgKnK95XOyTQzxyww3tZA/edit#gid=1784784202&range=K7)
> Export this payment record to a financial system if connected properly. Setting inherited from invoice

Reply:
Thank you for pointing me.